jetty-server/src/main/config/etc/jetty.xml

```diff
@@ -60,7 +60,7 @@
     <!-- =========================================================== -->
     <New id="httpConfig" class="org.eclipse.jetty.server.HttpConfiguration">
       <Set name="secureScheme" property="jetty.httpConfig.secureScheme"/>
-      <Set name="securePort" property="jetty.httpConfig.securePort"/>
+      <Set name="securePort"><Property name="jetty.httpConfig.securePort" default="8443"/></Set>
```

Comment:
I'm leery of this change.  
Isn't the in-code default `443`?

Reply:
Well the default is already 8443 in `server.ini`. The problem is that `<Set name="securePort" property="jetty.httpConfig.securePort"/>` will not actually set the property if it is not explicitly defined in the ini file.

In the `HttpConfiguration` class if securePort is not set is is by default a value of 0, and if the securePort value is 0 in `SecureRedirectHandler` it gives a 403 response saying `HttpConfiguration.securePort not configured`.

So to get the secure-redirect module working you would have to uncomment this manually in `server.ini`:
```
## Port to use to build URIs for secure redirects
# jetty.httpConfig.securePort=8443
```

Reply:
I agree that we should not put the default here, as that will change how we do defaults.

Instead we should modify the handler to be able to handle a 0 correctly, as I'm sure we already do in the SecurityHandler somewhere.  Hmmm no it also does a 403 if it sees 0 - so that definitely means we should not change the default.

Either they need to set the default port OR we can introduce some logic that 0 means redirect 80 to 443, but 403 response otherwise.